### PR TITLE
MWL-557 - add linting rules

### PR DESCRIPTION
**Pull request by roubaobaozi**

### CODE REVIEW

**Review by seankoji — APPROVED** [review]
weeeeeew

**Review by dompainter — APPROVED** [review]

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+insert_final_newline = true
+
+[*.html]
+indent_size = 4
+
+[*.html.twig]
+indent_size = 4
+
+[*.json]
+indent_style = tab
```
Comment (archinal) [inline .editorconfig]: What, really?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "eslint-config-foxsports",
+  "version": "0.0.0",
```
Comment (archinal) [inline package.json:3]: The [semver 2.0 spec](https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase) recommends starting at 0.1.0

File: src/configs/browserify-mocha-sinon-chai.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            property: 'only',
+        }, {
+            object: 'context',
+            property: 'only',
+        }, {
+            object: 'it',
+            property: 'only',
+        }, {
+            object: 'describe',
+            property: 'skip',
```
Comment (archinal) [inline src/configs/browserify-mocha-sinon-chai.js:46]: We're throwing errors for skipped tests now? Not just warnings?

File: src/configs/browserify-mocha-sinon-chai.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        }, {
+            object: 'context',
+            property: 'skip',
+        }, {
+            object: 'it',
+            property: 'skip',
+        }],
+        'object-curly-spacing': 'off',       // ignore object spacing, again typically JSON has spaces. We do not. But it's okay!
+        'quote-props':          'off',       // ignore quotes on props, fairly common for JSON responses in tests ok!
```
Comment (archinal) [inline src/configs/browserify-mocha-sinon-chai.js]: The space alignments on the values and comments are a bit off here

File: src/rules/stylistic-issues.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        'no-inline-comments': 'off',                                                              //  disallow inline comments after code
+        'no-lonely-if': 'error',                                                                  //  disallow if statements as the only statement in else blocks
+        'no-mixed-operators': 'error',                                                            //  disallow mixed binary operators
+        'no-mixed-spaces-and-tabs': 'error',                                                      //  disallow mixed spaces and tabs for indentation
+        'no-multi-assign': 'off',                                                                 //  disallow Use of Chained Assignment Expressions
+        'no-multiple-empty-lines': ['error', {max: 1}],                                           //  disallow multiple empty lines
+        'no-negated-condition': 'error',                                                          //  disallow negated conditions
+        'no-nested-ternary': 'error',                                                             //  disallow nested ternary expressions
```
Comment (archinal) [inline src/rules/stylistic-issues.js]: We seem to use (and override) this quite a bit now. Do we still want this rule to error?

File: src/rules/stylistic-issues.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        'no-continue': 'off',                                                                     //  disallow continue statements
+        'no-inline-comments': 'off',                                                              //  disallow inline comments after code
+        'no-lonely-if': 'error',                                                                  //  disallow if statements as the only statement in else blocks
+        'no-mixed-operators': 'error',                                                            //  disallow mixed binary operators
+        'no-mixed-spaces-and-tabs': 'error',                                                      //  disallow mixed spaces and tabs for indentation
+        'no-multi-assign': 'off',                                                                 //  disallow Use of Chained Assignment Expressions
+        'no-multiple-empty-lines': ['error', {max: 1}],                                           //  disallow multiple empty lines
+        'no-negated-condition': 'error',                                                          //  disallow negated conditions
```
Comment (archinal) [inline src/rules/stylistic-issues.js:85]: I've never been a fan of this one

File: .editorconfig (hunk 1 above)
Comment (roubaobaozi) [reply .editorconfig]: I think our .editorconfig has always been this way. Martian widgets etc. also have this. Do we want to change it?

File: src/configs/browserify-mocha-sinon-chai.js (hunk 3 above)
Comment (roubaobaozi) [reply src/configs/browserify-mocha-sinon-chai.js:46]: https://github.com/FoxSportsAustralia/fs-default-project-config/blob/master/resources/.eslintrc-standards--test-override.js#L19-L20 we always did 😄

File: .editorconfig (hunk 1 above)
Comment (archinal) [reply .editorconfig]: Well our package.json files use 2 spaces, so I don't think this value is really being respected. Regardless, should definitely be space IMO

Comment (roubaobaozi) [reply .editorconfig]: line 24 shows package.json has a different ruleset, for 2 spaces, not tabs. Same as previously. Which I always hated, too.

**Review by archinal — APPROVED** [review]

File: src/rules/stylistic-issues.js (hunk 5 above)
Comment (seankoji) [reply src/rules/stylistic-issues.js]: I would be super happy to turn this off and see if devs can use them responsibly >_>

File: .editorconfig (hunk 1 above)
Comment (seankoji) [reply .editorconfig]: yeah not sure where that started. we could update it from [.editorconfig](https://github.com/FoxSportsAustralia/fs-default-project-config/blob/master/resources/.editorconfig) and stop copying it around

**Pull request merged**